packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -843,7 +843,7 @@ class RenderOpacity extends RenderProxyBox {
        super(child);
 
   @override
-  bool get alwaysNeedsCompositing => child != null && (_alpha != 0 && _alpha != 255);
+  bool get alwaysNeedsCompositing => child != null && (_alpha != 0);
```

Comment:
While we're here, should this be `> 0`? Some places that check alpha use an inequality and other places use equality tests (see RenderAnimatedOpacityMixin below).

Reply:
I see that there are some asserts in here, but not sure if they protect every value test or not.

Reply:
Fair, I'll fix this. Previously we couldn't use that because 255 was also special cased :)